src/Symfony/Component/HttpFoundation/Tests/ResponseTest.php

```diff
@@ -659,6 +659,19 @@ public function testSetCache()
 
         $response->setCache(['immutable' => false]);
         $this->assertFalse($response->headers->hasCacheControlDirective('immutable'));
+
+        $directives = ['proxy_revalidate', 'must_revalidate', 'no_cache', 'no_store', 'no_transform'];
+        foreach ($directives as $directive) {
+            $response->setCache([$directive => true]);
+
+            $this->assertTrue($response->headers->hasCacheControlDirective(str_replace('_', '-', $directive)));
```

Comment:
we don't need the str_replace here, do we? (same below)

Reply:
fixed :+1:

Reply:
why fixed? the str_replace is still here :)

Reply:
`response->headers->hasCacheControlDirective` uses `-` ( correct way ), but setCache uses `_`, changing `setCache` to use `-` as well would be a BC-break.

Reply:
> changing setCache to use - as well would be a BC-break.

would it?

Reply:
This looks inconsistent to me, shouldn't we improve this?

Reply:
setCache(['last_modified' => $value]); wouldn't work anymore since it only works with `last-modified`, unless we reindex the directives array and replace all _ with -, this way we can support both.

Reply:
OK, unrelated anyway.